rs/registry/canister-client/src/lib.rs

```diff
@@ -93,4 +108,17 @@ pub trait CanisterRegistryClient: Send + Sync {
     /// over multiple messages.  It should generally be scheduled in a timer, but if it's never called
     /// the local registry data will not be in sync with the data in the Registry canister.
     async fn sync_registry_stored(&self) -> Result<RegistryVersion, String>;
+
+    /// Returns all keys that start with `key_prefix` and exist in between
+    /// `lower_bound` and `upper_bound`. Both `lower_bound` and `upper_bound`
+    /// are included in the returned result.
+    ///
+    /// The returned list does not contain any duplicates. There are no
+    /// guarantees wrt. the order of the contained elements.
+    fn get_effective_records_between(
+        &self,
+        key_prefix: &str,
+        lower_bound: RegistryVersion,
+        upper_bound: RegistryVersion,
+    ) -> Result<BTreeMap<StorableRegistryKey, StorableRegistryValue>, RegistryClientError>;
```

Comment:
It should not return "StorableRegistryKey" and "StorableRegistryValue" - these types should be internal to the stable storage implementation.  Isn't there a Registry type that would work here?

Reply:
I think RegistryVersionedRecord has all the same info, so it might be a good fit for the API as a vector.  I think ordering maybe should be guaranteed if the API returns a vec, however.

Reply:
Migrated to that approach, I think we don't have to guarantee ordering, although since the underlying data structure is a BTreeMap and we just transform it to Vec after traversal I believe we will have everything sorted.